### PR TITLE
feat(toolbar-demos): added toolbar back to demos

### LINKED ISSUE #3040: Add toolbar and pagination to demos
In https://github.com/patternfly/patternfly/pull/3039 we removed the legacy toolbar component from the following demos that we should add back

* Removes the toolbar & pagination from card view demo
  * Basic toolbar and pagination
* Removes toolbar & pagination from data-list demo
  * Basic toolbar and pagination all demos
  * no pagination demo
  * compact pagination demo
* Removes toolbar & pagination from table demos
  * Basic toolbar & pagination on all demos
  * compact pagination demo
  * column management dropdown
  * column management action

cc @mcarrano

### PR DESCRIPTION
**Pull request by mattnolting**
closes #3040 

**Previews:**
- `data-list` - https://patternfly-pr-3753.surge.sh/components/data-list/html-demos
- `table` - https://patternfly-pr-3753.surge.sh/components/table/html-demos
- `primary-detail` - https://patternfly-pr-3753.surge.sh/demos/primary-detail
- `card-view` - https://patternfly-pr-3753.surge.sh/components/card/html-demos

This PR includes a configurable `toolbar` template. It is usable across all components and demos.

### CODE REVIEW

**Review by mceledonia — APPROVED** [review]

**Review by mcarrano — APPROVED** [review]
Looks great @mattnolting .

**Comment by mcoker** [comment]
This looks great to me. Only thing I'm not sure about is that we added back the space around data-lists/tables in the page section, and that space persists on mobile so you get this space around the list/table like this

Screen Shot 2020-12-17 at 3 31 10 PM

I'm assuming we don't want that? Also related there is a bug in the page component where using a `pf-m-no-padding` class on a page section will remove the padding from page sections, even if you've added it back with a `pf-m-padding{-on[bp]}` class. I think we'll need to fix that to be able to remove the padding from mobile, but add it back on desktop.

**Comment by mcoker** [comment]
Confirmed with design that the demos should use page sections w/ no-padding/padding-on-xl by default, and that we can add a demo for both data-list and table that shows a full page component (section w/ no-padding)

**Comment by mcoker** [comment]
https://github.com/patternfly/patternfly/pull/3769/ was merged, so we can update the no-padding/padding-on-xl class on the page section in the table/data-list demos and this PR should be good to merge.

**Comment by mattnolting** [comment]
@mcoker updated the PR w/your comments. I also added `divider`s to match Sketch design files.

Screen Shot 2021-01-05 at 10 43 32 AM
Screen Shot 2021-01-05 at 10 42 53 AM

File: src/patternfly/demos/DataList/examples/DataList.md [hunk 1]
```diff
@@ -24,8 +24,10 @@ wrapperTag: div
     {{#> page-main-section page-main-section--modifier="pf-m-light"}}
       {{> data-list-main-section-content}}
     {{/page-main-section}}
-    {{#> page-main-section page-main-section--modifier="pf-m-no-padding pf-m-padding-on-md"}}
+    {{#> page-main-section}}
       {{#> card}}
+        {{> toolbar--template toolbar--id=(concat page--id '-toolbar') toolbar--template--HasBulkSelect="true" toolbar--template--HasContextSelector="true" toolbar--template--HasOverflowMenu="true"}}
+        {{> divider divider--type="div" divider--modifier="pf-u-hidden pf-u-display-flex-on-lg"}}
```
Comment (mcoker) [inline src/patternfly/demos/DataList/examples/DataList.md:30]: The data-list has a top border, so this creates a double border.

**Review by mcoker — APPROVED** [review]
LPTM!

**Pull request merged**